docs/source/weightvec.rst

```diff
@@ -47,16 +47,37 @@ Let ``w`` be an instance of ``WeightVec``:
 
 .. function:: sum(w)
 
-    Get the sum of weights. 
+    Get the sum of weights.
 
     :note: The sum of weights is maintained by the weight vector, and thus this function can immediately return the value in ``O(1)`` (without computation).
 
 
-Why we want a WeightVec type
------------------------------
+Why we want an AbstractWeights type
+------------------------------------
 
-The ``WeightVector`` type is introduced as the standard way to pass weights, which has two advantages:
+The ``AbstractWeights`` type is introduced as the standard way to pass weights, which has two advantages:
 
-- A different type ``WeightVec`` distinguishes the role of the weight vector from other data vectors in the input arguments.
+- A different type ``AbstractWeights`` distinguishes the role of the weight vector from other data vectors in the input arguments.
 - Statistical functions that utilize weights often need the sum of weights for various purposes. The weight vector maintains the sum of weights, so that it needn't be computed repeatedly each time the sum of weights is needed.
 
+
+Other AbstractWeights types and bias correction
```

Comment:
"other" doesn't make a lot of sense here since no other type has been introduced. It would be nice to have a section giving a short summary of the four concrete types of weights (mostly copy/paste of the docstrings). Then this section could just be called "Variance bias correction", and it could also mention that no correction is possible for `Weights` because it isn't specific enough.

BTW, needs backticks around `AbstractWeights` even in headings IIRC.